xdsl/interpreter.py

```diff
@@ -726,11 +733,11 @@ def cast_value(self, o: Attribute, r: Attribute, value: Any) -> Any:
     def value_for_attribute(self, attr: Attribute, type_attr: Attribute) -> Any:
         return self._impls.attr_value(self, attr, type_attr)
 
-    def get_op_for_symbol(self, symbol: str) -> Operation:
-        if symbol in self.symbol_table:
-            return self.symbol_table[symbol]
-        else:
-            raise InterpretationError(f'Could not find symbol "{symbol}"')
+    def get_op_for_symbol(self, symbol: str | SymbolRefAttr) -> Operation:
+        op = SymbolTable.lookup_symbol(self.module, symbol)
```

Comment:
This is not exactly the same lookup, as the interpreter maintains a lookup via the `symbol_table` property.
When using the `SymbolTable.lookup_symbol` you indicate the operation from where to start the search and move "outwards"; here this seems to go straight to the containing module.

Not sure why something like this wouldn't do the trick? (haven't tested it):

```py
if isinstance(symbol, SymbolRefAttr):
    symbol = symbol.data
if symbol in self.symbol_table:
    return self.symbol_table[symbol]
...
```

Reply:
Hmm, I see; that lookup as you described happens via the `SymbolTable::lookup`; to me then sounds like the intepreter lookup needs a fix

Reply:
Is there a way to get the current operation being handled by the interpreter, some sort of program counter. If not I can add it and use that value instead of `self.module` to seed the lookup?

Reply:
This definitely seems like an upgrade in functionality and we need to figure out how to fix riscv to make it work properly with these symbols. The existing symbol table helper should be deleted IMO and just replaced with this, the clients might as well call it directly, and give the relevant root op. I can see how to refactor RISC-V here to work better, and that seems like a pre-requisite to merging this.